Spanner/src/Database.php

```diff
@@ -871,6 +872,10 @@ public function runTransaction(callable $operation, array $options = [])
         };
 
         $delayFn = function (\Exception $e) {
+            $defaultRetryDelay = ['seconds' => 0, 'nanos' => 0];
+            if ($e->getCode() === Code::INTERNAL && strpos($e->getMessage(), 'RST_STREAM') !== false) {
```

Comment:
Should we really be hardcoding this check like this?

If it's a `ServerException` then doing this inside of a method in the particular exception could be reasonable.
The definition does the same for an `AbortedException`, WDYT?

Also, nit: if this delay array has context in multiple exceptions, then it can be taken out to the base exception class, i.e. `ServiceException`.

Reply:
I have moved the delay to `ServiceException` since it seemed like a better location.
https://github.com/googleapis/google-cloud-php/pull/5955, once this is merged I will update this PR with the core version.